src/webview/create-component/createComponentLoader.ts

```diff
@@ -0,0 +1,156 @@
+    }
+
+    static async loadView(title: string, component: CompTypeDesc, starterProjectName: string): Promise<vscode.WebviewPanel> {
+        const webViewTitle = component ?  `${title}-${component.devfileData.devfile.metadata.displayName}` : title;
+        const localResourceRoot = vscode.Uri.file(path.join(CreateComponentLoader.extensionPath, 'out', 'componentViewer'));
+        panel = vscode.window.createWebviewPanel('CreateComponentView', webViewTitle, vscode.ViewColumn.One, {
+            enableScripts: true,
+            localResourceRoots: [localResourceRoot],
+            retainContextWhenHidden: true
+        });
+        panel.iconPath = vscode.Uri.file(path.join(CreateComponentLoader.extensionPath, 'images/gitImport/git.svg'));
+        panel.webview.html = CreateComponentLoader.getWebviewContent(CreateComponentLoader.extensionPath, panel);
+        panel.onDidDispose(() => {
+            panel = undefined;
+        });
+        panel.webview.onDidReceiveMessage(createComponentMessageListener);
+        if (component && starterProjectName) {
+            panel?.webview.postMessage({
+                action: 'InputFromDevFile',
+                selectedComponent: component,
+                selectedPro: starterProjectName
+            });
+        }
+        return panel;
+    }
+
+    private static getWebviewContent(extensionPath: string, p: vscode.WebviewPanel): string {
+        // Local path to main script run in the webview
+        const reactAppRootOnDisk = path.join(extensionPath, 'out', 'componentViewer');
+        const reactAppPathOnDisk = vscode.Uri.file(
+            path.join(reactAppRootOnDisk, 'componentViewer.js'),
+        );
+        const reactAppUri = p.webview.asWebviewUri(reactAppPathOnDisk);
+        const htmlString: Buffer = fs.readFileSync(path.join(reactAppRootOnDisk, 'index.html'));
+        const meta = `<meta http-equiv="Content-Security-Policy"
+            content="connect-src *;
+            default-src 'none';
+            img-src ${p.webview.cspSource} https: 'self' data:;
+            script-src 'unsafe-eval' 'unsafe-inline' vscode-resource:;
+            style-src 'self' vscode-resource: 'unsafe-inline';">`;
+        return `${htmlString}`
+            .replace('%COMMAND%', '')
+            .replace('%PLATFORM%', process.platform)
+            .replace('componentViewer.js', `${reactAppUri}`)
+            .replace('%BASE_URL%', `${reactAppUri}`)
+            .replace('<!-- meta http-equiv="Content-Security-Policy" -->', meta);
+    }
+
+    static refresh(): void {
+        if (panel) {
+            panel?.webview.postMessage({ action: 'loadingComponents' });
+        }
+    }
+}
+
+function validateName(event: any) {
+    let validationMessage = event.action !== 'validateAppName' ? OpenShiftItem.emptyName(`Required ${event.name}`, event.name.trim()) : null;
+    if (!validationMessage) validationMessage = OpenShiftItem.validateMatches(`Not a valid ${event.name}.
```

Comment:
`event.name` refers to the text in the input box. I think it would make more sense if you wrote something like

`${event.name} is not a valid ${validateAppName ? 'app' : 'component'} name.`